lib/OpenStackExtract/MiqOpenStackVm/MiqOpenStackCommon.rb

```diff
@@ -50,6 +53,55 @@ def get_image_file_glance_v2(image_id)
     end
     tf
   end
+  
+  def create_image_from_snapshot(snapshot_id, disk_format)
+    snapshot = volume_service.snapshots.get(snapshot_id)
+    volume_options = {
+      :name        => "Temp Volume from #{snapshot.name}",
+      :size        => snapshot.size,
+      :snapshot_id => snapshot_id
+    }
+    volume = volume_service.volumes.new(volume_options)
+    volume.save
+    while volume.status != 'available'
+      sleep(10)
+      volume = volume_service.volumes.get(volume.id)
+    end
+    response = volume_service.action(volume.id, 'os-volume_upload_image' => {
+                                       :image_name => "Temp Image from #{snapshot.name}",
+                                       :disk_format => disk_format})
+    image_id = response.body["os-volume_upload_image"]["image_id"]
+    while image_service.images.get(image_id).status.downcase != 'active'
+      sleep(10)
+    end
+    return image_service.images.get(image_id)
+  ensure
+    volume.destroy
```

Comment:
If we fail before the line:
```ruby
volume = volume_service.volumes.new(volume_options)
```
then `volume` won't be set when we enter the ensure block, causing a runtime error.
The `ensure` block should be associated with a `begin` block that starts after the `volume =` line.